godot_code/start_window.gd

```diff
@@ -0,0 +1,35 @@
+extends VBoxContainer
+
+@onready var menuPlace: Control = $Columns/MenuCol
+var lobbyNode: VBoxContainer = null
+
+signal stateChanged
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	menuPlace.get_child(0).connect("startButtonPressed", openGameJoin)
+
+func newPlayerJoins(clientId: String, playerName: String):
+	lobbyNode.newPlayerJoins(clientId, playerName)
+
+func openGameJoin():
+	var lobbyScene = preload("res://player_lobby.tscn")
+	for child in menuPlace.get_children():
+		child.queue_free()
+	lobbyNode = lobbyScene.instantiate()
+	lobbyNode.connect("backToMain", openMainMenu)
+	menuPlace.add_child(lobbyNode)
+	emit_signal("stateChanged", "joinGame")
+
+func openMainMenu():
+	var menuScene = preload("res://main_menu.tscn")
+	for child in menuPlace.get_children():
+		child.queue_free()
+	var menuNode: VBoxContainer = menuScene.instantiate()
+	lobbyNode = null
+	menuNode.connect("startButtonPressed", openGameJoin)
+	menuPlace.add_child(menuNode)
+	emit_signal("stateChanged", "mainMenu")
+
+func _on_exit_button_pressed():
```

Comment:
When the game runs through the console software it runs as a child process. The console software kills the the child process when it receives an exit message so the game itself should not handle the exit

Reply:
I guess we can just close the game window to simulate that kind of exit process.